gst/tensor_decoder/tensordec.c

```diff
@@ -1004,37 +1011,26 @@ gst_tensordec_set_caps (GstBaseTransform * trans,
   silent_debug_caps (incaps, "from incaps");
   silent_debug_caps (outcaps, "from outcaps");
 
-  if (g_strcmp0 (self->mode, "image_labeling") == 0) {
-
-    GstStructure *structure;
-    GstTensorsConfig config;
-
-    /* Need to update in cases video or text */
-    gst_base_transform_set_in_place (trans, FALSE);
-  /** compare output tensor */
-    structure = gst_caps_get_structure (outcaps, 0);
-    gst_tensors_config_from_structure (&config, structure);
-    self->negotiated = TRUE;
-
-    return TRUE;
-
-  } else {
   /** compare and verify outcaps */
-    if (gst_tensordec_configure (self, incaps)) {
-      GstTensorConfig config;
-      GstStructure *s = gst_caps_get_structure (outcaps, 0);
+  if (gst_tensordec_configure (self, incaps)) {
+    GstTensorConfig config;
+    GstStructure *s = gst_caps_get_structure (outcaps, 0);
 
-      if (gst_tensor_config_from_structure (&config, s) &&
-          gst_tensordec_check_consistency (self, &config)) {
-        self->negotiated = TRUE;
-      }
+    if (gst_tensor_config_from_structure (&config, s) &&
+        gst_tensordec_check_consistency (self, &config)) {
+      self->negotiated = TRUE;
     }
-
+  }
+  if (self->mode == IMAGE_LABELING) {
+    gst_base_transform_set_in_place (trans, FALSE);
+    self->negotiated = TRUE;
```

Comment:
Please remove this line.
(this flag 'negotiated' is set in line 1021. if negotiated is false, plugin caps negotiation is failed.)

Reply:
for this problem,
I added 	(config->rate_n = 30; config->info.dimension[0] = 1001;)
without this, text caps negotiation fails

Reply:
for this problem,
I added 	(config->rate_n = 30; config->info.dimension[0] = 1001;)
without this, text caps negotiation fails

Reply:
for this problem,
I added 	(config->rate_n = 30; config->info.dimension[0] = 1001;)
without this, text caps negotiation fails